elasticgraph-graphql/spec/unit/elastic_graph/graphql/datastore_query/shard_routing_spec.rb

```diff
@@ -259,6 +259,24 @@ def shard_routing_for(route_with_field_paths, filter_or_filters)
         })).to search_all_shards
       end
 
+      it "searches all shards when we have an `any_of: [{anyof: []}]` filter because that will match all results" do
+        expect(shard_routing_for(["name"], {
+          "any_of" => [{"any_of" => []}]
+        })).to search_all_shards
```

Comment:
I think we can search no shards in this case, because (in contrast to the test description), `"any_of" => [{"any_of" => []}]` should match no documents, right?

That said, it doesn't produce incorrect results to search all shards when we can search none.  So if this is hard to implement correctly, it's ok to err on the side of searching all shards.